yjit/src/codegen.rs

```diff
@@ -4549,8 +4549,29 @@ fn gen_send_cfunc(
         return CantCompile;
     }
 
-    if flags & VM_CALL_ARGS_SPLAT != 0  {
-        gen_counter_incr!(asm, send_args_splat_cfunc);
+    if flags & VM_CALL_ARGS_SPLAT != 0 && cfunc_argc == -1 {
+        gen_counter_incr!(asm, send_args_splat_cfunc_negative);
+        return CantCompile;
+    }
```

Comment:
What does it mean when `cfunc_argc == -1` ? Does it mean vararg ?

Should explain in comments and give `send_args_splat_cfunc_negative` a more descriptive name.

Reply:
Yeah, I'll change that. Good callout